### PR TITLE
GitHub action test build jobs

### PR DESCRIPTION
**Pull request by layko-andrey**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      MONGODB_AUTH_DB: admin
+
+    services:
+      mongodb:
+        image: mongo:4.4.0
+        ports:
+          - 27017:27017
+        env:
+          MONGO_INITDB_ROOT_USERNAME: open_state
+          MONGO_INITDB_ROOT_PASSWORD:  open_state
```
Comment (trenatov) [inline .github/workflows/deploy.yml]: format please

**Review by trenatov — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
+name: Deploy Open State
+
+on:
+  push:
+    branches: [ 2.x.x-REACTIVE ]
```
Comment (ruslanys) [inline .github/workflows/deploy.yml]: Is there a chance to take a look at the pipeline before the merge?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+jobs:
+
+  tests:
+    name: Run tests
+    runs-on: ubuntu-latest
+    continue-on-error: false
+
+    env:
```
Comment (ruslanys) [inline .github/workflows/deploy.yml]: I don't think we need to duplicate default values as we already have them in `application.properties` file.

**Review by ruslanys — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (ruslanys) [reply .github/workflows/deploy.yml]: BTW, we have to run tests at all of the branches. We can do `build` on the protected branches (`master` / `2.x.x-REACTIVE`) and on PR (the section below)

Comment (layko-andrey) [reply .github/workflows/deploy.yml]: No. We have to merge these changes to 2.x.x.Release to see lit of workflows and have ability to run it. Workflow not visible from pull request unfortunately.

Comment (layko-andrey) [reply .github/workflows/deploy.yml]: I would check it, as far as I know GitHub has ability to run Actions on any selected branch

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (layko-andrey) [reply .github/workflows/deploy.yml]: Fixed, thanks

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (layko-andrey) [reply .github/workflows/deploy.yml]: But we can change it at any time according our needs. But workflow would be independent  of these cases and tests would run in own environment space.

Comment (ruslanys) [reply .github/workflows/deploy.yml]: For the test cases, there is no need to do this. If we have specific parameters, we will define it here. Until it is the same as we have in the application.properties, there are no reasons to duplicate the code.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (ruslanys) [reply .github/workflows/deploy.yml]: For sure we can. You just need to define more (or even all) branches instead.

Comment (layko-andrey) [reply .github/workflows/deploy.yml]: Ok, would be done

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (layko-andrey) [reply .github/workflows/deploy.yml]: Would fix it

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
+name: Deploy Open State
```
Comment (ruslanys) [inline .github/workflows/deploy.yml]: Change the name

**Comment by ruslanys** [comment]
1. Change the name of the Workflow
2. We have to start the pipelines on more then just 2.x.x-REACTIVE branch.
2.1. * (all branches): test
2.2. [protected branches] (master / 2.x.x-REACTIVE): all above, + build -> jar

**Comment by ruslanys** [comment]
The next PR should:

! extend `build` with docker packaging

--

* deploy
* code coverage (codecov.io)

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (layko-andrey) [reply .github/workflows/deploy.yml]: Fixed

**Comment by layko-andrey** [comment]
> 1. Change the name of the Workflow
> 2. We have to start the pipelines on more then just 2.x.x-REACTIVE branch.
>    2.1. * (all branches): test
>    2.2. [protected branches] (master / 2.x.x-REACTIVE): all above, + build -> jar

**Review by ruslanys — APPROVED** [review]

**Review by ruslanys — APPROVED** [review]

**Pull request merged**